enhancements/security/security-profiles-operator.md

```diff
@@ -0,0 +1,636 @@
+
+The SPOD instance is configurable through a CRD which allows us to set where
+it's scheduled and the capabilities that are enabled/disabled. For OpenShift,
+SELinux would be enabled by default and the DaemonSet would be scheduled
+everywhere.
+
+`SeccompProfiles` are handled by simply persisting a file on a directory that
+the Kubelet has access to. A `SeccompProfile` looks as follows:
+
+```yaml
+---
+apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
+kind: SeccompProfile
+metadata:
+  name: profile-complain-block-high-risk
+  annotations:
+    description: "Enables complain mode whilst blocking high-risk syscalls. Some essential syscalls are allowed to decrease log noise."
+spec:
+  defaultAction: SCMP_ACT_LOG
+  architectures:
+  - SCMP_ARCH_X86_64
+  syscalls:
+  - action: SCMP_ACT_ALLOW
+    names:
+    - exit
+    - exit_group
+    - futex
+    - nanosleep
+
+  - action: SCMP_ACT_ERRNO
+    names:
+    - acct
+    - add_key
+    - bpf
+    - clock_adjtime
+    - clock_settime
+    - create_module
+    - delete_module
+    - finit_module
+    - get_kernel_syms
+    - get_mempolicy
+    - init_module
+    - ioperm
+    - iopl
+    - kcmp
+    - kexec_file_load
+    - kexec_load
+    - keyctl
+    - lookup_dcookie
+    - mbind
+    - mount
+    - move_pages
+    - name_to_handle_at
+    - nfsservctl
+    - open_by_handle_at
+    - perf_event_open
+    - personality
+    - pivot_root
+    - process_vm_readv
+    - process_vm_writev
+    - ptrace
+    - query_module
+    - quotactl
+    - reboot
+    - request_key
+    - set_mempolicy
+    - setns
+    - settimeofday
+    - stime
+    - swapoff
+    - swapon
+    - _sysctl
+    - sysfs
+    - umount2
+    - umount
+    - unshare
+    - uselib
+    - userfaultfd
+    - ustat
+    - vm86old
+    - vm86
+```
+
+As seen, the object is simply a wrapper around the JSON formatting, which
+allows us to do extra validations on the objects, thus ensuring easily that
+it's appropriately formatted.
+
+`SelinuxProfiles` require extra work, as to install a SELinux policy
+one has to parse the policy, compile it to the main system's policy
+and then install it in the system (which keeps it in memory). To do this
+an extra component called [selinuxd](https://github.com/JAORMX/selinuxd) has
+been developed. The intent is that selinuxd allows one to simply drop a file
+on a pre-determined directory to install a SELinux policy, and that policy will
+get picked up and installed by the daemon. Thus, the SPOD instance simply
+places the policy into a directory, the selinuxd instance picks it up
+and handles the installation and tracking of policies.
+
+A `SelinuxProfile` object looks as follows:
+
+```yaml
+---
+apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
+kind: SelinuxProfile
+metadata:
+  name: errorlogger
+  namespace: my-namespace
+spec:
+  policy: |
+    (blockinherit container)
+    (allow process var_log_t ( dir ( open read getattr lock search ioctl add_name remove_name write ))) 
```

Comment:
Is their guidance on how a customer can evaluate whether the SELinux policy they are installing going to make their cluster less secure?

Reply:
Is there such guidance for writing SELinux policy in general? We're working on a SELinux policy recording feature for SPO, would it help to suggest that, as opposed to hand-craft the policies so that the policy would only ever allow what the app is doing (presumably in CI) ?

Reply:
@mrunalp could you specify what you mean by guidance?